lib/isolator/orm_adapters/disconnection_handler.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Isolator
+  class DisconnectionHandler
+    def self.register_cleanup!
+      # Add callback to clean up connection after db disconnection.
+      connection_adapter_klasses = ActiveRecord::Base.connection_handler.connection_pools.map { |pool| pool.connection.class }.uniq
+      connection_adapter_klasses.each do |connection_adapter_klass|
+        connection_adapter_klass.set_callback :checkin, :after do |conn|
```

Comment:
According to [this source](https://stackoverflow.com/questions/21446094/a-callback-for-activerecord-database-connections) and [this rails code](https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L14) it looks like `checkin` callback should be invoked on disconnection. But I have not been able to see the callback invoked for the right connection.